kotlin-source/src/main/kotlin/net/corda/training/contract/IOUContract.kt

```diff
@@ -84,14 +84,10 @@ class IOUContract : Contract {
                     requireThat { "There must be one output IOU." using (ious.outputs.size == 1) }
                     // Check only the paid property changes.
                     val outputIou = ious.outputs.single()
-                    requireThat {
-                        "The amount may not change when settling." using (inputIou.amount == outputIou.amount)
-                        "The borrower may not change when settling." using (inputIou.borrower == outputIou.borrower)
-                        "The lender may not change when settling." using (inputIou.lender == outputIou.lender)
-                    }
+                    requireThat { "Only the paid amount can change." } using (inputIou.copy(paid = outputIou.paid) == outputIou)
```

Comment:
While this is logically correct sometimes it is better to keep things simpler on multiple lines in order to help better drive education. Having three lines shows how each field can be checked independently in a real use case.

Reply:
@henrikr3 asked me about this as well on the other PR. The reason I made this change was to try and recommend a design pattern that is more future proof. Checking only what can be changed and returning that hopefully adds two benefits:

1. The end user understands the fields that can change instead of getting a message that says you can't do what you are trying to do, which seems only half useful.
2. If new fields are added you don't have to change the contract. Whereas the old checks all other fields are not being changed individually so you would have to change the contract when changing the state. Maybe overkill but the same pattern was used here - https://github.com/corda/corda-training-solutions/blob/7edbcc5cc144ad57f53320e1682fcfd6c1a5671c/kotlin-source/src/main/kotlin/net/corda/training/contract/IOUContract.kt#L57

Reply:
Although I used copy instead of the state function in the check linked above. Again I think this decouples the test from the code but I'm not going to get too pedantic about it :)

Reply:
You're 100% right for future proofing a normal application. The worry is that we see many skill levels in training and there is a risk of over-complicating the code. It is good to teach best practices but sometimes folks aren't ready.

PR is approved either way. I'd recommend teaching a course to get a feel for how the content works.